designdocs/provider_contexts.md

```diff
@@ -0,0 +1,230 @@
+
+    #[pallet::storage]
+	pub type ProviderToApplicationRegistryEntry<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		ProviderId,
+		Twox64Concat,
+		ApplicationIdentifier<T>,
+		ApplicationRegistryEntry<T::MaxProviderNameSize, T::MaxProviderLogo250X100Size>,
+		OptionQuery,
+	>;
+	```
+5. `MaxProviderNameSize` be increased to `256`.
+6. `MaxProviderLogo250X100Size` be created and the limit set to `131_072` (128 KiB).
+
+### **Mainnet Approval Flow** <a id='governance'></a>
+
+_Any_ change to the Provider or Application Context must be approved by governance.
+For now, that governance approval is any single Frequency Council member.
+
+```mermaid
+sequenceDiagram
+    participant P as Provider
+    participant D as Provider Dashboard
+    participant F as Frequency
+    participant G as Frequency Council
+
+    note over P,G: Provider Registration
+    P->>D: Login
+    D-->>+F: Create MSA
+    F-->>-D: MSA created
+    P->>D: Request to be Provider form
+    P->>D: Add Applications as needed
+    D-->>+F: Generate Frequency Council Proposal
+    G->>F: Review and Approve Proposal
+    F->>F: Execute Provider registration
+    F-->>-D: MSA is a Provider
+
+    note over P,G: Provider Update Registration
+    P->>D: Login
+    P->>D: See Provider and Application Information
+    P->>D: Request to update Provider form
+    P->>D: Update/add Applications as needed
+    D-->>+F: Generate Frequency Council Proposal
+    G->>F: Review and Approve Update Proposal
+    F->>F: Execute Provider changes as an upsert
+    F-->>-D: Update display
+    P->>D: See Updated Provider and Application Information
+```
+
+#### Open Questions
+- Is this a set or edit pattern for the applications?
+- Should adding/updating applications be a separate call?
+- Should adding/updating a translation be a separate call?
+
+#### Implementation Suggestion
+
+To limit the amount of unapproved bytes intended to be interpreted as images on chain, an option would be to merely submit the hashes of images for approval.
+This hash would then be allowlisted for that provider.
+Once approved, that provider could submit the image to chain.
+In the governance process, a link to the image would need to be submitted.
+
+### **Example of Wallet Usage via SIWF** <a id='siwf'></a>
+
+SIWF [Signed Request Payload](https://projectlibertylabs.github.io/siwf/v2/docs/DataStructures/All.html) can expand the optional `applicationContext` value with a new optional `id` field that is the text of the `ApplicationIdentifier` on chain.
+This would allow a smooth transition between `applicationContext.url` and `applicationContext.id` with both being optional.
+
+The Wallet would then:
+1. Verify the SIWF Signed Request.
+2. Lookup the Provider via the `publicKey` in the SIWF Signed Request.
+3. If any, fetch the Application Identifier from Frequency.
+4. Display the information from the ApplicationRegistryEntry (or default Provider one) to the user to help them know who they are authorizing.
```

Comment:
Can we better specify what is meant by the "default Provider one"?

Reply:
```suggestion
4. Display the information from the `ApplicationRegistryEntry` (or the Provider Registry Entry `ProviderToRegistryEntry` if there is no application context identifier) to the user to help them know who they are authorizing.
```